### PR TITLE
Video support in PubMatic adapter

### PR DESCRIPTION
**Pull request by PubMatic-OpenWrap**
## Type of change

- [x ] Feature

## Description of change

Adding support for video in PubMatic adapter.

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/prebid/prebid.github.io/pull/865

### CODE REVIEW

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -18,6 +19,22 @@ const CUSTOM_PARAMS = {
   'profId': '', // OpenWrap Legacy: Profile ID
   'verId': '' // OpenWrap Legacy: version ID
 };
+const VIDEO_CUSTOM_PARAMS = {
+  'mimes': 'array',
```
Comment (pm-harshad-mane) [inline modules/pubmaticBidAdapter.js]: move data-type to constant

File: modules/pubmaticBidAdapter.js [hunk 2]
```diff
@@ -152,27 +169,90 @@ function _createOrtbTemplate(conf) {
   };
 }
 
+// similar functionality as parseSlotParam. Check if the 2 functions can be clubbed.
+function _checkParamDataType(key, value, datatype) {
+  var errMsg = 'PubMatic: Ignoring param key: ' + key + ', expects ' + datatype + ', found ' + typeof value;
+  switch (datatype) {
+    case 'boolean':
```
Comment (pm-harshad-mane) [inline modules/pubmaticBidAdapter.js]: use data-type constant

File: modules/pubmaticBidAdapter.js [hunk 3]
```diff
@@ -287,7 +381,17 @@ export const spec = {
               referrer: utils.getTopWindowUrl(),
               ad: bid.adm
             };
-
+            var parsedRequest = JSON.parse(request.data);
```
Comment (jaiminpanchal27) [inline modules/pubmaticBidAdapter.js]: Use `let` at all places

File: src/constants.json [hunk 4]
```diff
@@ -68,5 +68,11 @@
     "SRC" : "s2s",
     "DEFAULT_ENDPOINT" : "https://prebid.adnxs.com/pbs/v1/openrtb2/auction",
     "SYNCED_BIDDERS_KEY": "pbjsSyncs"
+  },
+  "DATA_TYPES": {
```
Comment (jaiminpanchal27) [inline src/constants.json]: You can add these to your adapter.

**Pull request merged**